compiler/rustc_codegen_llvm/src/declare.rs

```diff
@@ -147,7 +147,7 @@ impl<'ll, 'tcx> CodegenCx<'ll, 'tcx> {
                 for options in [
                     TypeIdOptions::GENERALIZE_POINTERS,
                     TypeIdOptions::NORMALIZE_INTEGERS,
-                    TypeIdOptions::NO_SELF_TYPE_ERASURE,
+                    TypeIdOptions::ERASE_SELF_TYPE,
```

Comment:
I don't know, I kinda disagree with the change to the NO_SELF_TYPE_ERASURE. I think the default should be what is going to be done most of the time or more commonly, and it's performing type erasure. The exception for the secondary type id should be opted in when needed.

Reply:
I don't have a strong preference - I like this version a little better due to the "positive" nature of it.

With regards to defaults vs secondary, only KCFI has a meaningful notion of a "default" type ID - CFI attaches all of them to the same attachment site, so none of them are really primary or secondary. KCFI still sets `ERASE_SELF_TYPE` as its default with this changeset.

That said, if this change is contentious and not required by compiler-errors to land this PR, I don't really care to fight over this.

Reply:
It's secondary because these are not always included. They're only included when they're unique and for methods, and also are only tested for when methods are used as function pointers so the concept of secondary does apply generally.